be/src/fs/hdfs/hdfs_fs_cache.cpp

```diff
@@ -49,6 +49,16 @@ static const std::map<std::string, std::string> get_cloud_properties(const FSOpt
 
 static Status create_hdfs_fs_handle(const std::string& namenode, const std::shared_ptr<HdfsFsClient>& hdfs_client,
                                     const FSOptions& options) {
+    // we don't force user to set JAVA_HOME.
+    // but when we access hdfs via JNI, we have to make sure JAVA_HOME is set,
+    // if not BE will crash because of failure to create JVM.
+    {
+        const char* p = std::getenv("JAVA_HOME");
```

Comment:
possibly reduce the call with `call_once`?

Reply:
I think it's not good
1. check cost is very low.
2. we expect to return Status here, but call once only return void.

Reply:
prefer check it in ExecEnv

Reply:
> prefer check it in ExecEnv

We only need to check when we use JNI. Sometimes we don't use JNI feature at all.

Reply:
And jni is used at two places(or two entries)
- libhdfs
- java_udf

Reply:
> I think it's not good
> 
> 1. check cost is very low.
> 2. we expect to return Status here, but call once only return void.

I mean, we can check only once for all the lifecycle and save the check result, as the check result is not expected to change once the process is started. For later checks, just return the cached result.

Reply:
and you are right, the cost is low, so it is optional. your call to check it every time or just once. I am all good.